### PR TITLE
OOP exercise

### PR DESCRIPTION
**Pull request by Shrai-dev**
### OOP exercise

[Demo](https://shrai-dev.github.io/a-tiny-JS-world/)
[Code base](https://github.com/Shrai-dev/a-tiny-JS-world)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

File: submissions/Shrai-dev/oop-exercise/index.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+   }
+
+}
+
+// ======== OUTPUT ========
+
+const dog = new Animal ('dog', 'Buddy', 'male', 3, 4, 'bark-bark!', ['Elsa', 'Chip', 'Max']);
+const cat = new Animal ('cat', 'Chloe', 'female', 2, 4, 'meow!', ['Leo', 'Simon', 'Kitty']);
+const man = new Human ('human', 'Brian', 'male', 26, 2, 2, 'Everyone thinks of changing the world, but no one thinks of changing himself!', ['Ethan', 'James', 'Ryan']);
+const woman = new Human ('human', 'Olivia', 'female', 24, 2, 2, 'The simplest way to be happy is to do good!', ['Maria', 'Kelly', 'Jennifer']);
```
Comment (OleksiyRudenko) [inline submissions/Shrai-dev/oop-exercise/index.js:52]: Sending a species string value to a species class looks like unnecessary duplication.
Well, `Human` is a species. `Animal` is not. If you had `Dog` and `Cat` classes you could avoid this for them too.

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@Shrai-dev good job!
However there is inconsistency in hierarchy of classes. Please fix.
Making hierarchy consistent will help you to improve the solution further.

File: submissions/Shrai-dev/oop-exercise/index.js (hunk 1 above)
Comment (Shrai-dev) [reply submissions/Shrai-dev/oop-exercise/index.js]: @OleksiyRudenko Thanks for pointing to that. I've fixed these duplications.

File: submissions/Shrai-dev/oop-exercise/index.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+   }
+}
+
+class Cat extends Animal {
+  constructor(name, gender, age, friends) {
+    super("cat", name, gender, age, 4, "meow!", friends);
+  }
+  toString() {
+    return super.toString();
+  }
```
Comment (OleksiyRudenko) [inline submissions/Shrai-dev/oop-exercise/index.js:40]: Every method is inherited just like properties.
If the only purpose of a method overloading is to call an inherited method, then there is no actually point.

File: submissions/Shrai-dev/oop-exercise/index.js [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+}
+
+class Woman extends Human {
+   constructor(name, age, saying, friends) {
+      super("human", name, "female", age, 2, 2, saying, friends);
+   }
+toString() {
+   return super.toString();
+}
```
Comment (OleksiyRudenko) [inline submissions/Shrai-dev/oop-exercise/index.js:71]: Same as above.

File: submissions/Shrai-dev/oop-exercise/index.js [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+}
+
+class Man extends Human {
+   constructor(name, age, saying, friends) {
+     super("human", name, "male", age, 2, 2, saying, friends);
+   }
+   toString() {
+     return super.toString();
+   }
```
Comment (OleksiyRudenko) [inline submissions/Shrai-dev/oop-exercise/index.js:80]: Same as above.

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@Shrai-dev well done!
Let's polish the code.

**Review by OleksiyRudenko — APPROVED** [review]
@Shrai-dev great job!

**Pull request merged**